cmip6_downscaling/workflows/bcsd_flow.py

```diff
@@ -1,116 +1,125 @@
-
-with Flow(name=flow_name) as flow:
-    gcm = run_hyperparameters["GCM"]
-    scenario = run_hyperparameters["SCENARIO"]
-    train_period_start = run_hyperparameters["TRAIN_PERIOD_START"]
-    train_period_end = run_hyperparameters["TRAIN_PERIOD_END"]
-    predict_period_start = run_hyperparameters["PREDICT_PERIOD_START"]
-    predict_period_end = run_hyperparameters["PREDICT_PERIOD_END"]
-    variable = run_hyperparameters["VARIABLE"]
-
-    coarse_obs_path, spatial_anomalies_path, bias_corrected_path, final_out_path = make_flow_paths(
-        **run_hyperparameters
+write_bcsd_results_task = task(write_bcsd_results, log_stdout=True)
+
+
+# Main Flow -----------------------------------------------------------
+
+# with Flow(name="bcsd-testing", storage=storage, run_config=run_config) as flow:
+# with Flow(name="bcsd-testing", storage=storage, run_config=kubernetes_run_config, executor=dask_executor) as flow:
+with Flow(name="bcsd-testing") as flow:
+
+    flow_name = Parameter("FLOW_NAME")
+    gcm = Parameter("GCM")
+    scenario = Parameter("SCENARIO")
+    train_period_start = Parameter("TRAIN_PERIOD_START")
+    train_period_end = Parameter("TRAIN_PERIOD_END")
+    predict_period_start = Parameter("PREDICT_PERIOD_START")
+    predict_period_end = Parameter("PREDICT_PERIOD_END")
+    variable = Parameter("VARIABLE")
+
+    (
+        coarse_obs_path,
+        spatial_anomalies_path,
+        bias_corrected_path,
+        final_out_path,
+    ) = make_flow_paths_task(
+        GCM=gcm,
+        SCENARIO=scenario,
+        TRAIN_PERIOD_START=train_period_start,
+        TRAIN_PERIOD_END=train_period_end,
+        PREDICT_PERIOD_START=predict_period_start,
+        PREDICT_PERIOD_END=predict_period_end,
+        VARIABLE=variable,
     )
-
-    coarse_obs_path, spatial_anomalies_path = preprocess_bcsd(
-        gcm=gcm,
-        train_period_start=train_period_start,
-        train_period_end=train_period_end,
-        variable=variable,
-        coarse_obs_path=coarse_obs_path,
-        spatial_anomalies_path=spatial_anomalies_path,
-        connection_string=connection_string,
-        rerun=True,
+    # preprocess_bcsd_tasks(s):
+    obs_ds = return_obs_task(train_period_start, train_period_end, variable)
+    coarse_obs_ds = get_coarse_obs_task(obs_ds, variable, connection_string)
+    spatial_anomalies_ds = get_spatial_anomalies_task(
+        coarse_obs_ds, obs_ds, variable, connection_string
     )
 
-    y_rechunked_path, X_train_rechunked_path, X_predict_rechunked_path = prep_bcsd_inputs(
-        coarse_obs_path=coarse_obs_path,
-        gcm=gcm,
-        scenario=scenario,
-        train_period_start=train_period_start,
-        train_period_end=train_period_end,
-        predict_period_start=predict_period_start,
-        predict_period_end=predict_period_end,
-        variable=variable,
+    # prep_bcsd_inputs_task(s):
+    y_rechunked_ds = return_y_rechunked_task(coarse_obs_ds, variable)
```

Comment:
instead of "rechunked", what do you think about naming these variables as how they're chunked? e.g. `full_space`, etc